compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -67,10 +71,34 @@ fn intern_shallow<'rt, 'mir, 'tcx, T, M: CompileTimeMachine<'mir, 'tcx, T>>(
     }
     // link the alloc id to the actual allocation
     let alloc = ecx.tcx.mk_const_alloc(alloc);
-    ecx.tcx.set_alloc_id_memory(alloc_id, alloc);
+    if let Some(static_id) = ecx.machine.static_def_id() {
+        intern_as_new_static(ecx.tcx, static_id, alloc_id, alloc);
+    } else {
+        ecx.tcx.set_alloc_id_memory(alloc_id, alloc);
+    }
     Ok(alloc.0.0.provenance().ptrs().iter().map(|&(_, prov)| prov))
 }
 
+/// Creates a new `DefId` and feeds all the right queries to make this `DefId`
+/// appear as if it were a user-written `static` (though it has no HIR).
```

Comment:
Is "feeding" a query a standard term? I haven't seen it before. What does it do? Set the reply to the query as-if it had been computed that way?

What happens if we forget one query in this?

Reply:
Yes, this is an established concept in rustc.

If you forget, the query body is executed, which ICEs, because there is no hir. If a query does not depend on HIR, it can succeed